xdsl/utils/exceptions.py

```diff
@@ -37,6 +37,53 @@ def __str__(self) -> str:
                f"arguments {self.args}"
 
 
+class DeferredExceptionMessage:
+    """
+    This class can be used to defer exception message generation to the
+    time when the message is printed.
+
+    We saw a problem with our parser where parse times in specific
```

Comment:
Nit, I would change this paragraph to remove the history of this class.
I would just describe the problem it is fixing, since in 1-2 years, people won't know about the old parser, so it's best to remove any mention of it :)